File: ios/PizzaDelivery/Sources/ViewModels/ChatbotViewModel.swift
import Foundation

@MainActor
class ChatbotViewModel: ObservableObject {
    @Published var messages: [ChatMessage] = []
    @Published var isLoading = false
    @Published var inputText = ""
    @Published var errorMessage: String?
    
    private let chatbotService = ChatbotService.shared
    private var webSocketService: WebSocketService?
    
    init() {
        messages.append(ChatMessage(
            content: "Hi! I'm your Pizza Assistant. How can I help you today? I can help with menu recommendations, order tracking, or answer any questions!",
            role: .assistant,
            suggestions: ["Show me today's specials", "Track my order", "What's popular?"]
        ))
    }
    
    func sendMessage(_ text: String? = nil) async {
        let messageText = text ?? inputText
        guard !messageText.trimmingCharacters(in: .whitespaces).isEmpty else { return }
        
        let userMessage = ChatMessage(content: messageText, role: .user)
        messages.append(userMessage)
        inputText = ""
        isLoading = true
        
        do {
            let response = try await chatbotService.sendMessage(message: messageText)
            let assistantMessage = ChatMessage(
                id: response.id,
                content: response.message,
                role: .assistant,
                suggestions: response.suggestions
            )
            messages.append(assistantMessage)
        } catch {
            let errorMsg = ChatMessage(
                content: "Sorry, I'm having trouble connecting. Please try again.",
                role: .assistant
            )
            messages.append(errorMsg)
            errorMessage = error.localizedDescription
        }
        
        isLoading = false
    }
    
    func connectWebSocket() {
        webSocketService = WebSocketService(url: AppConfig.WebSocket.chatbot)
        webSocketService?.connect()
    }
    
    func disconnectWebSocket() {
        webSocketService?.disconnect()
    }
}

File: android/app/src/main/java/com/pizzadelivery/android/ui/chatbot/ChatbotViewModel.kt
package com.pizzadelivery.android.ui.chatbot

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.pizzadelivery.android.data.model.ChatMessage
import com.pizzadelivery.android.data.model.ChatRole
import com.pizzadelivery.android.data.repository.ChatRepository
import com.pizzadelivery.android.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import java.util.UUID
import javax.inject.Inject

@HiltViewModel
class ChatbotViewModel @Inject constructor(
    private val chatRepository: ChatRepository
) : ViewModel() {

    private val _messages = MutableStateFlow<List<ChatMessage>>(emptyList())
    val messages: StateFlow<List<ChatMessage>> = _messages.asStateFlow()

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()

    private var conversationId: String? = null

    init {
        // Add initial greeting
        _messages.value = listOf(
            ChatMessage(
                id = UUID.randomUUID().toString(),
                content = "Hi! I'm your Pizza Delivery assistant. How can I help you today? I can help with:\n• Menu recommendations\n• Order status\n• Store information\n• Delivery updates",
                role = ChatRole.ASSISTANT,
                timestamp = System.currentTimeMillis().toString(),
                suggestions = listOf("Show me the menu", "Track my order", "Find nearest store", "Show offers")
            )
        )
    }

    fun sendMessage(content: String) {
        if (content.isBlank()) return

        val userMessage = ChatMessage(
            id = UUID.randomUUID().toString(),
            content = content,
            role = ChatRole.USER,
            timestamp = System.currentTimeMillis().toString()
        )

        _messages.value = _messages.value + userMessage
        _isLoading.value = true

        viewModelScope.launch {
            when (val result = chatRepository.sendMessage(content, conversationId)) {
                is Resource.Success -> {
                    result.data?.let { response ->
                        conversationId = response.conversationId
                        _messages.value = _messages.value + response.message
                    }
                }
                is Resource.Error -> {
                    val errorMessage = ChatMessage(
                        id = UUID.randomUUID().toString(),
                        content = "Sorry, I'm having trouble connecting. Please try again.",
                        role = ChatRole.ASSISTANT,
                        timestamp = System.currentTimeMillis().toString()
                    )
                    _messages.value = _messages.value + errorMessage
                }
                is Resource.Loading -> {}
            }
            _isLoading.value = false
        }
    }
}
